### PR TITLE
Update villagerAutoTrader.sc

### PR DESCRIPTION
**Pull request by Ghoulboy78**
Honestly, It's mostly the ReadMe, I thought I had changed more code than that on my own copy at https://github.com/Ghoulboy78/scripts/blob/master/Finished%20ideas/villagerAutoTrader.sc ...
Anyways, this says it all imo.

### CODE REVIEW

**Review by BisUmTo — APPROVED** [review]

File: programs/survival/villager_auto_trader.sc [hunk 1]
```diff
@@ -1,27 +1,57 @@
...
+            _:'buy':'id'=='minecraft:emerald' &&//only emerald trades
+            _:'buyB':'id'=='minecraft:air' &&//don't want second trade item
+            _:'maxUses'!=0 //Cos if not it's cheaty, dont work rn
+        );
+        if(!trade, return())//again, this is a cheap call and only called once
+    );
+
+    if(nbt(zombie_villager~'nbt':'HandItems'):'[]':0:'id'=='minecraft:emerald' && (hand_count=nbt(zombie_villager~'nbt':'HandItems'):'[]':0:'Count') >= ((cost=trade:'buy':'Count')),
+
+        modify(zombie_villager,'nbt_merge',str('{HandItems:[{id:"minecraft:emerald",Count:%sb}]}',hand_count-cost));
```
Comment (Firigion) [inline programs/survival/villager_auto_trader.sc:29]: Just out of curiosity: didn't `invenotry_set` work here?

File: programs/survival/villager_auto_trader.sc [hunk 2]
```diff
@@ -1,27 +1,57 @@
...
+
+        item = null;
+        in_dimension(zombie_villager,
+            item_pos = pos(zombie_villager)+[0, zombie_villager~'eye_height'-0.3, 0]; 
+            item = spawn('item',item_pos,str('{Item:{id:"%s",Count:%db, PickupDelay:10}}',trade:'sell':'id',trade:'sell':'Count'));//todo add velocity towards player
```
Comment (Firigion) [inline programs/survival/villager_auto_trader.sc:34]: regarding the commen on this line: you added it already :P

File: programs/survival/villager_auto_trader.sc [hunk 3]
```diff
@@ -1,27 +1,57 @@
...
+            distance_sq = _euclidean_sq(item_pos, pos(_));
+            if(distance_sq<distance_sq_to_nearest_player,
+                nearest_player = _;
+                distance_sq_to_nearest_player = distance_sq
+            )
+        );
+
+        if(nearest_player!=null,
+            item_motion = (pos(nearest_player)-item_pos)*0.3;
+            modify(item, 'motion', item_motion)
```
Comment (Firigion) [inline programs/survival/villager_auto_trader.sc:50]: Upon testing, this is waaaay to fast motion for an item. Maybe change it to whatever vanilla does for bartering?

File: programs/survival/villager_auto_trader.sc [hunk 4]
```diff
@@ -1,27 +1,57 @@
...
+
+        for(entity_area('player', item_pos, [5,5,5]),
+            distance_sq = _euclidean_sq(item_pos, pos(_));
+            if(distance_sq<distance_sq_to_nearest_player,
+                nearest_player = _;
+                distance_sq_to_nearest_player = distance_sq
+            )
+        );
+
+        if(nearest_player!=null,
```
Comment (Firigion) [inline programs/survival/villager_auto_trader.sc:48]: If no payer is around, i think the item should still be thrown in the zombie's look direction.

**Review by Firigion — APPROVED** [review]
No critical flaw or bug, just minor comments and suggestions. take them or leave them, if you don't wanna take 'em, merge 'em.

File: programs/survival/villager_auto_trader.sc (hunk 1 above)
Comment (Ghoulboy78) [reply programs/survival/villager_auto_trader.sc:29]: no, i'm afraid it didn't

**Pull request merged**